ucx.spec.in

```diff
@@ -62,6 +62,7 @@ BuildRequires: xpmem-devel
 %endif
 %if %{with java}
 BuildRequires: maven
+BuildRequires: java-1.8.0-openjdk-devel
```

Comment:
do we really want hardcoded java version here?
maybe check distro version?

Reply:
IMO need specify as
```
BuildRequires:  java-openjdk-devel > 1.8.0
```

Reply:
UCX's `m4` scripts should be updated to support JDK > 1.8.0, currently, they fail to detect it automatically and require `$JAVA_HOME` to be set.

Reply:
@amaslenn what is the problem with m4 scripts?

Reply:
Consider the following OS configuration (similar to what is present in [Fedora's Koji](https://kojipkgs.fedoraproject.org//work/tasks/303/43680303/root.log) system):
```sh
# rpm -qa | grep '^java'
java-latest-openjdk-devel-14.0.0.36-3.rolling.fc30.x86_64
javapackages-filesystem-5.3.0-4.fc30.noarch
java-1.8.0-openjdk-headless-1.8.0.222.b10-0.fc30.x86_64
java-latest-openjdk-14.0.0.36-3.rolling.fc30.x86_64
java-latest-openjdk-headless-14.0.0.36-3.rolling.fc30.x86_64
javapackages-tools-5.3.0-4.fc30.noarch
```

UCX's JAVA auto-detection gets wrong path to JAVA:
```sh
# ./autogen.sh && ./configure --with-java
...
checking for tc_malloc in -ltcmalloc... no
checking for mvn... yes
checking for java... yes
checking for readlink... yes
configure: WARNING: Please set JAVA_HOME=/usr/lib/jvm/java-1.8.0-openjdk-1.8.0.222.b10-0.fc30.x86_64
checking jni_md.h usability... no
checking jni_md.h presence... no
checking for jni_md.h... no
configure: error: Java support requested, but couldn't find jni headers in /usr/lib/jvm/java-1.8.0-openjdk-1.8.0.222.b10-0.fc30.x86_64
```

The correct one should be `/usr/lib/jvm/java-14/` I guess. The detected path doesn't contain devel files.

The main issue here is that we have several Java versions, but only one has devel files.

Anyway, I'm not an expert, @petro-rudenko could you please comment?

Reply:
Ok here's an issue. If no `JAVA_HOME` is set, autoconf trying to find it by itself, by running `readlink java`: https://github.com/openucx/ucx/blob/1f45e05b168b4e8b15421d6b54d0583d28ebbf41/config/m4/java.m4#L29

You have both java14 installed (JDK) and java8 (just runtime, no jni headers). Default java on the system is java8. So solution is just set JAVA_HOME=/path/to/java14

Reply:
In other words, we ask the user to specify their Java version (which is OK). In our case, the user is rpm spec + rpm build, which is version agnostic. I see only the following options:
1. Make UCX's Java detection smarter.
2. Implement Java detection in SPEC files (not sure if it is possible).

I'd vote for (1), the implementation for both cases will be the same, but (1) will also help `./configure` users.

@petro-rudenko @yosefe ?

Reply:
Can you please try #5093

Reply:
is it possible older distros will not have it?
pls consider that UCX is built on a variety of distros in MLNX_OFED for example
also, what about ubuntu/debian?

Reply:
This is a rather old version, I guess all distros in use have it. Ubuntu/Debian is not affected since it is a spec for RPM. Ubuntu usually has newer versions, so it brings up a compilation issue, but it is unrelated to this PR.